src/coreclr/jit/lowerxarch.cpp

```diff
@@ -7503,6 +7501,50 @@ bool Lowering::IsContainableHWIntrinsicOp(GenTreeHWIntrinsic* parentNode, GenTre
             return false;
         }
 
+        case NI_SSE3_MoveAndDuplicate:
+        case NI_AVX2_BroadcastScalarToVector128:
+        case NI_AVX2_BroadcastScalarToVector256:
+        case NI_AVX512F_BroadcastScalarToVector512:
+        {
+            // make the broadcast node containable when embedded broadcast can be enabled.
+            if (intrinsicId == NI_SSE3_MoveAndDuplicate)
+            {
+                // NI_SSE3_MoveAndDuplicate is for Vector128<double> only.
+                assert(childNode->AsHWIntrinsic()->GetSimdBaseType() == TYP_DOUBLE);
+            }
+            if (comp->compOpportunisticallyDependsOn(InstructionSet_AVX512F_VL) &&
+                parentNode->OperIsEmbBroadcastCompatible())
+            {
+                GenTree* broadcastOperand = childNode->AsHWIntrinsic()->Op(1);
+                bool     childSupportsRegOptional;
+                if (broadcastOperand->OperIs(GT_LCL_VAR))
+                {
+                    const unsigned opLclNum = broadcastOperand->AsLclVar()->GetLclNum();
+                    comp->lvaSetVarDoNotEnregister(opLclNum DEBUGARG(DoNotEnregisterReason::LiveInOutOfHandler));
```

Comment:
I don't think we want to do this since it may negatively impact other users of the same local number.

We only have two special places where we set `DoNotEnregister` in lowering and that's for `struct args` as part of `PUTARG_STK` (ABI handling) and a pessimization of `StoreBlkSrc` since it can't work with enregistered values today.

Rather the existing `IsContainableHWIntrinsicOp` checks should already be limiting this to only allow containment if its already been marked as `DoNotEnregister`, which should happen for the most important cases.

-- It would be good to get secondary weigh in from one of the JIT team members on this

Reply:
Seems like the PR got updated after your comment, but I generally agree with the principle: "it may negatively impact other users of the same local number".

Reply:
Yes, I can understand the point, the related parts has been deleted.